packages/vega-parser/src/parsers/legend.js

```diff
@@ -150,6 +152,11 @@ function buildLegendEncode(_, config) {
     strokeDash:   config.strokeDash,
     x:            _('legendX'),
     y:            _('legendY'),
+    ariaHidden:   ariaHidden,
```

Comment:
Same issues with ariaHidden-conditional logic noted previously. Since this pattern is repeated, perhaps it would be better to abstract these to a dedicated aria helper, and pass the config object through that helper (which could then annotate the config object).

Reply:
Maybe something like this?

```js
function addAriaAnnotations(spec, _, props) {
  const hidden = _(props.hidden);
  const aria = hidden === true
    ? {
        ariaHidden: hidden
      }
    : {
        ariaHidden: hidden,
        ariaLabel: _(props.label),
        ariaRole: _(props.role),
        ariaRoleDescription: _(props.roleDescription),
        tabindex: _(props.tabindex)
      };
      
  return Object.assign(spec, aria);
}
```

Reply:
Done. I needed a bit more for the function to generate the default label.

Reply:
I honestly got confused by the code. The legend code uses `addEncoders` while the axis code uses `extendEncode` and expects every property to be passed through and `encoder`.

Can you help me?

Reply:
I didn't really know what do with the encoder so I'm just passing it as an argument. This feels wrong. Can you help me?

Reply:
I'm using the shorthand for titles and subtitles now. I couldn't figure out a clean way to make it work for guides since their implementations differ (to me) significantly.